### ISSUE TITLE
Sidebar + Sticky menu on same page

### ISSUE THREAD

**Issue by tecoad** [Semantic-Org/Semantic-UI#1745]
Hi there,

I'm trying to use a sidebar and a sticky menu on the same page. 
However, when scrolling down and clicking to view sidebar, the menu detaches from the top when sidebar is visible.
The behavior I would expect in this case would be the menu to slide with the content when sidebar is visible, as it does when using a fixed top menu.

Is it possible? Am i doing it right?
Check fiddles:

Sticky menu + Sidebar: http://jsfiddle.net/tecoad/g2sy72mf/6/
Fixed top menu + Sidebar (expected behavior): http://jsfiddle.net/tecoad/g2sy72mf/5/

**Comment by ph7vc** [comment]
Everything you want to be "pushed" must be inside the `pusher` div.

http://jsfiddle.net/438s86on/

Works as expected.

Also with the sticky stuff, you should specify a `context`

**Comment by tecoad** [comment]
@ph7vc 
no. the example you provided works exactly as mine. note that after scrolling, menu doesn't slide with the content when sidebar is visible.
**how it is**:
![captura de tela 2015-02-02 as 20 51 20](https://cloud.githubusercontent.com/assets/2627749/6010864/605a8066-ab1d-11e4-888f-311e5ce0714b.png)

**how it should be**:
![captura de tela 2015-02-02 as 20 50 49](https://cloud.githubusercontent.com/assets/2627749/6010867/64634a94-ab1d-11e4-909e-fe5deedff9a2.png)

**Comment by ph7vc** [comment]
Here is what I see. Looks fine to me

![semantic-pusher-bug](https://cloud.githubusercontent.com/assets/5940454/6011010/195d1bee-aafd-11e4-913e-835a245829e7.png)

**Comment by tecoad** [comment]
This is freakin odd. This is how the same page appears to me on Chrome for Mac:

![captura de tela 2015-02-02 as 21 28 21](https://cloud.githubusercontent.com/assets/2627749/6011407/8b2b9262-ab22-11e4-8c9a-fee3a911e114.png)

**Comment by jlukic** [comment]
Sidebars are a pain for maintaining coordinate systems. I discuss this in the release notes, and in some other threads. The spec is ambiguous about how to handle translate and coords.

The only transition that does not *break* position fixed, is `overlay` which does not modify the coordinate system of `pusher`

You can see an example of this with sticky here:
http://jsfiddle.net/k8nsy6xb/

**Comment by lauri-elevant** [comment]
Just for your information, I managed to solve my issues with Sidebar + Sticky by just moving the sticky element out of the pushed tree, directly to document.body. Being placed there, sticky starts working nicely. Note that 260px is the width for my sidebar, YMMV.

	$('.following.menu').sticky({
		onStick: function() {
			if ($(this).parents('.pushed').length) {
				$(this).data('displaced', $(this).parent());
				$(this).detach().appendTo(document.body).css('left', '-=260px');
			}
		},
		onUnstick: function() {
			if ($(this).data('displaced')) {
				$(this).detach().appendTo($(this).data('displaced')).data('displaced', null);
			}
		}
	});

You probably also need to attach something like that to the sidebar code:

		onShow: function() {
			$('.following.menu').each(function() { $(this).data('moduleSticky').refresh(); });
		},
		onHidden: function() {
			$('.following.menu').each(function() { $(this).data('moduleSticky').refresh(); });
		}

@jlukic  Could the sticky be moved out to body by default? I do not see any problems with this approach - the sticky (especially while being "stuck") is absolutely positioned anyway..

If you approve, I can put in some time for a PR.

**Comment by jlukic** [comment]
This is a sore spot in the css spec, `translate3d` adds the unexpected consequence of causing a new stacking context w/r/t `position: fixed`.  I've played this issue out in the Chromium bug tracker before, but it needs to be resolved on a w3c spec level.

**Comment by lauri-elevant** [comment]
@jlukic

Fully aware of the spec situation. This is why I propose simply bypassing all of that and putting the (already abs. positioned) element outside of the ```translate3d```-affected DOM tree. Do you see any downsides? Maybe make a separate option, like in popups code?

It fixed my issues, anyway. Just wanted to check if you think it is worth coding this approach to SUI directly. Since I did not have to resort to hacking SUI, it is not an issue for me any more, but I can put in some time if it would align with your plans.

Thanks for the syntax pointer! :-) I was getting too comfortable working with the internal stuff ;)

And keep up the awesome work! :+1:  :+1:  :+1:

**Comment by jlukic** [comment]
I'm not sure the exact consequences of this, but I would believe there would be some issues depending on the layout structure. I'll open a separate issue.

**Comment by andrew-oko-odion** [comment]
Any Fix for this yet?
I am using React Semantic-ui and its a major problem

**Comment by andrew-oko-odion** [comment]
> Just for your information, I managed to solve my issues with Sidebar + Sticky by just moving the sticky element out of the pushed tree, directly to document.body. Being placed there, sticky starts working nicely. Note that 260px is the width for my sidebar, YMMV.
> 
> ```
> $('.following.menu').sticky({
>     onStick: function() {
>         if ($(this).parents('.pushed').length) {
>             $(this).data('displaced', $(this).parent());
>             $(this).detach().appendTo(document.body).css('left', '-=260px');
>         }
>     },
>     onUnstick: function() {
>         if ($(this).data('displaced')) {
>             $(this).detach().appendTo($(this).data('displaced')).data('displaced', null);
>         }
>     }
> });
> ```
> You probably also need to attach something like that to the sidebar code:
> 
> ```
>     onShow: function() {
>         $('.following.menu').each(function() { $(this).data('moduleSticky').refresh(); });
>     },
>     onHidden: function() {
>         $('.following.menu').each(function() { $(this).data('moduleSticky').refresh(); });
>     }
> ```
> @jlukic Could the sticky be moved out to body by default? I do not see any problems with this approach - the sticky (especially while being "stuck") is absolutely positioned anyway..
> 
> If you approve, I can put in some time for a PR.
@lauri-elevant  I am trying to do the same, using semantic-ui-react and its not working please can you show a example please?
